app/models/user.rb

```diff
@@ -0,0 +1,31 @@
+require 'bcrypt'
+
+class User
+  include DataMapper::Resource
+
+  attr_reader :password
+  attr_accessor :password_confirmation
+  validates_confirmation_of :password
+  validates_presence_of :email
+  validates_uniqueness_of :email
```

Comment:
Prefer the new style validations `validates :column, uniqueness: value` over `validates_uniqueness_of`.